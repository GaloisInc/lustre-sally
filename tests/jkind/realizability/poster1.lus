-- Unrealizable without assert
-- Realizable with assert

node main(in, out : int) returns (ok : bool);
let
  -- assert in >= 0;
  ok = (out = 2 * in) and (out >= 0);
  --%PROPERTY ok;
  --%REALIZABLE in;
tel;
