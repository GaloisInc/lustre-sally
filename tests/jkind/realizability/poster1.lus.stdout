[Lustre] Loading model from: "poster1.lus"
Sally Model
===========

(define-state-type
   S
   (
     (in Int)
     (ok Bool)
     (out Int)
     (|gal-initializing| Bool))
   (
     (in Int)
     (out Int)))
(define-transition-system
   TS
   S
   (= |gal-initializing| true)
   (and
      (= next.|gal-initializing| false)
      (= next.in input.in)
      (= next.out input.out)
      (=
         next.ok
         (and
            (=
               next.out
               (* 2 next.in))
            (<= 0 next.out)))))

(query
   TS
   (or |gal-initializing| ok))

[Lustre] Validating properties:
[Lustre] Property ok...
  [Sally][Invalid]
[Trace] 
ok:
 Step | in | out | |-> | ok    
------+----+-----+-----+-------
 1    | -1 | -1  |     | False 

[Lustre] Summary:
  [Valid] 0
  [Unknown] 0
  [Invalid] 1
[Lustre] Model status: [Invalid]
